### PR TITLE
Refactoring error transaction_queue error handling and `update_sealing` method.

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

**Comment by tomusdrw** [comment]
Because I've updated `update_sealing` pending transactions are now not displayed unless you are actually mining.

How do we want it to work? We could ditch `sealing_enabled` if we want to display pending block transactions in all cases.

**Comment by gavofyork** [comment]
that's fine. if you're not mining, then you shouldn't be authoring new blocks. if you're not authoring new blocks then it questionable whether you should be running the transaction queue (and relaying transactions) at all.

File: sync/src/chain.rs [hunk 1]
```diff
@@ -950,7 +950,10 @@ impl ChainSync {
 			nonce: chain.nonce(a),
 			balance: chain.balance(a),
 		};
-		let _ = self.miner.import_transactions(transactions, fetch_account);
+		let res = self.miner.import_transactions(transactions, fetch_account);
+		if res.is_ok() {
```
Comment (gavofyork) [inline sync/src/chain.rs]: what's this about?

**Comment by tomusdrw** [comment]
Ok. So parity nodes in general will show `0` as pending transactions on ethstats - it's ok, right?

File: sync/src/chain.rs (hunk 1 above)
Comment (tomusdrw) [inline sync/src/chain.rs]: It updates the block only if all transactions were imported successfuly. It should rather be `if any transaction was imported successfuly`.

Comment (gavofyork) [inline sync/src/chain.rs]: actually, it should never update the block

**Comment by gavofyork** [comment]
unless they're mining, yes that's fine.

**Comment by arkpar** [comment]
needs merge

**Pull request merged**